cli/utils/api.go

```diff
@@ -85,13 +79,20 @@ func (resourceClient ResourceClient) NewRequest(url string, method string, body
 		return nil, err
 	}
 
-	request.Header = resourceClient.BaseHeader
+	if contentType == "" {
+		contentType = "application/json"
+	}
+
+	request.Header.Set("x-client-id", analytics.ClientID())
+	request.Header.Set("Content-Type", contentType)
+	request.Header.Set("Accept", contentType)
```

Comment:
No, because with that we are saying "we expect the same input and output contentType from the API", on future versions we have the idea of sending yaml and handling the output with the `--output-format` flag that the user uses.

Reply:
for future referece, if you don't need different request and response content types, you can pass just one (either) header and everything falls back to that

Reply:
Cool! I'm to revisit this decision and start to use different `Content-Type` and `Accept` after we visit the PR https://github.com/kubeshop/tracetest/pull/2523 .
In the future the main idea is that the backend should be responsible for formatting entities to avoid replication of the same logic and both CLI and Backend.